runtime/src/main/java/org/corfudb/protocols/wireprotocol/IMetadata.java

```diff
@@ -78,10 +79,30 @@ default Long getRank() {
      * @param rank The rank of this append.
      */
     default void setRank(long rank) {
+        if (rank>=0 && getRank()<0) {
+            throw new IllegalStateException("Cannot set rank when the overwrite is permitted");
```

Comment:
The idea was if you have after quorum replication of 3 replicas values 1 1 and 2, to send such write request to the replica that has 1.

Good to discuss this as there is another approach here - just when we are up to throw and overwrite exception to read the old rank and compare.
Extra read operation would not add extra cost, as overwrite exceptioon is rare, also there is no atomicity issue as you have a lock for the segment.

I could change it to this approach, but probably we will need to change Rank as long to Rang as Comparable or Rank as long+uuid. Do you think it is OK to have rank = long+uuid globally, not for quorum replication only. If yes, this solves all problems.